crates/rpc-replay/src/transactions.rs

```diff
@@ -90,21 +98,196 @@ pub fn starknet_rs_to_blockifier(
+                            sender_address: starknet_api::core::ContractAddress(PatriciaKey::try_from(
+                                tx.sender_address,
+                            )?),
+                        },
+                    );
+
+                    (tx_hash, api_tx, tx.class_hash)
+                }
+                DeclareTransaction::V3(tx) => {
+                    let tx_hash = TransactionHash(tx.transaction_hash);
+                    let api_tx = starknet_api::transaction::DeclareTransaction::V3(
+                        starknet_api::transaction::DeclareTransactionV3 {
+                            resource_bounds: resource_bounds_core_to_api(&tx.resource_bounds),
+                            tip: starknet_api::transaction::Tip(tx.tip),
+                            signature: starknet_api::transaction::TransactionSignature(tx.signature.clone()),
+                            nonce: starknet_api::core::Nonce(tx.nonce),
+                            class_hash: starknet_api::core::ClassHash(tx.class_hash),
+                            compiled_class_hash: starknet_api::core::CompiledClassHash(tx.compiled_class_hash),
+                            sender_address: starknet_api::core::ContractAddress(PatriciaKey::try_from(
+                                tx.sender_address,
+                            )?),
+                            nonce_data_availability_mode: da_mode_core_to_api(tx.nonce_data_availability_mode),
+                            fee_data_availability_mode: da_mode_core_to_api(tx.fee_data_availability_mode),
+                            paymaster_data: starknet_api::transaction::PaymasterData(tx.paymaster_data.clone()),
+                            account_deployment_data: starknet_api::transaction::AccountDeploymentData(
+                                tx.account_deployment_data.clone(),
+                            ),
+                        },
+                    );
+
+                    (tx_hash, api_tx, tx.class_hash)
+                }
+
+                _ => unimplemented!("DeclareTransaction V0 not supported"),
+            };
+
+            // TODO: improve this to avoid retrieving this twice. Already done in main.rs from prove_block
+            let starknet_contract_class = provider.get_class(BlockId::Number(block_number), class_hash).await?;
+            let generic_sierra_cc = match starknet_contract_class {
+                starknet::core::types::ContractClass::Sierra(flattened_sierra_cc) => {
+                    GenericSierraContractClass::from(flattened_sierra_cc)
+                }
+                starknet::core::types::ContractClass::Legacy(_) => {
+                    unimplemented!("Fixme: Support legacy contract class")
+                }
+            };
+
+            let flattened_sierra = generic_sierra_cc.clone().to_starknet_core_contract_class()?;
+            let contract_class = generic_sierra_cc.compile()?.get_blockifier_contract_class()?.clone();
```

Comment:
There's (or at least should be) a `to_blockifier_contract_class` method that hides the clone.